### PR TITLE
revision of the first few episodes of make-novice workshop

### PR DESCRIPTION
**Pull request by tomichec**
Hi,

First I would like to thank you for maintaining this course. I think Make is a great tool that many more people should know about.

I have revised the first few episodes of the course and send some changes. I hope you find them useful.

Besides that I have found some other places for possible improvement, but as they require more changes to be introduced they are likely to be more time-consuming. So before I do that I will submit a separate issue, so we can discuss it first.

I find the structure of the course really nice.

Please, let me know if there are some amendments you would like me to do on this PR.

Cheers,
Tomas

### CODE REVIEW

File: _episodes/04-dependencies.md [hunk 1]
```diff
@@ -103,24 +103,24 @@ python countwords.py books/last.txt last.dat
...
+The following figure shows a graph of the dependencies, that are
+involved in building the target `results.txt`. Notice the recently
+added dependencies `countwords.py` and `testzipf.py`.  This is how the
+Makefile should look like after completing the rest of the exercises
+in this episode.
 
 ![results.txt dependencies after adding countwords.py and testzipf.py as dependencies](../fig/04-dependencies.png "results.txt dependencies after adding countwords.py and testzipf.py as dependencies")
 
-> ## Why Don't the `.txt` Files Depend on `countwords.py`?
+> ## Why the `.txt` Files Don't Depend on `countwords.py`?
```
Comment (gcapes) [inline _episodes/04-dependencies.md:114]: This suggestion is grammatically incorrect - we should keep the original heading

**Review by gcapes — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute to this lesson!

I've now managed to go through this PR, and have added some comments to be addressed. Some great suggestions in here, but also some rewordings that I don't feel improve on the existing lesson.

File: _episodes/02-makefiles.md [hunk 2]
```diff
@@ -204,11 +193,15 @@ make: `isles.dat' is up to date.
...
-to build `abyss.dat`:
+This means that even though we have added yet another rule, the new
+target `abyss.dat` has not been build. What Make actually did when we
+run it without any command line arguments is to try to build its
+[default target]({{ page.root }}/reference.html#default-target) which
+is the first target it finds. Since this is `isles.dat`, which is
+already up-to-date it was not recreated.
+
+If we want to build a non-default target we need to specify it
+explicitely as the command line argument:
```
[the remark this answers is not in the document]
Comment (tomichec) [reply _episodes/02-makefiles.md:205]: How would you rephrase it to make it more clear?

My aim was to make it more explicit what is meant by "nothing happens", because actually some computation is indeed done and the result is that the target is not build.

File: _episodes/02-makefiles.md [hunk 3]
```diff
@@ -303,9 +296,12 @@ make: `clean' is up to date.
...
 
-Make finds a file (or directory) called `clean` and, as its `clean`
-rule has no dependencies, assumes that `clean` has been built and is
-up-to-date and so does not execute the rule's actions. As we are using
+What happened is that
+Make found a  directory called `clean` and, as its `clean`.
+Since the `clean` rule has no dependencies, it assumes that `clean` directory has already been built (and is
+up-to-date) and so it does not execute the rule's actions. 
+
+As we are using
```
[the remark this answers is not in the document]
Comment (tomichec) [reply _episodes/02-makefiles.md:305]: right, the wording is not completely clear. I will commit changes.

**Review by tomichec — COMMENTED** [review]
Hi @gcapes , thanks for the review. I am going to address your comments both here and send further commit with further changes. Feel free to include whatever you find useful and modify what you don't find appropriate.
Cheers,
T.
